### ISSUE TITLE
Several service tests broken by 'Revert "[vm/aot/tfa] Tree shake write-only fields"'

### ISSUE THREAD

**Issue by sjindel-google** [dart-lang/sdk#40637]
Broken tests:

```
service/contexts_test
service/get_retaining_path_rpc_test
service/instance_field_order_rpc_test
service/object_graph_vm_test
service/regexp_function_test/0
service/regexp_function_test/1
```

I've approved the results.

**Comment by alexmarkov** [comment]
These tests were failing on obfuscation bot before ff34fd81106dbd4c2a341f5545e5f900b591ebfa. ff34fd81106dbd4c2a341f5545e5f900b591ebfa fixed those tests, and reverting that change in 8ac7f6bce7d0619bf2404d58428508e79d9ce90b broke them again.

**Comment by alexmarkov** [comment]
These tests were fixed once again in 12c8b02c848aeb5fe5731533649319ba43a2e9b4.
